### PR TITLE
CashBox page

### PR DESCRIPTION
**Pull request by BarLisML**
Inserting a cashbox csv file, creating and finding a cashbox, transferring mintToken to cashbox in different batches
Changing variable names and handling amount using decimal in toTransfer func

### CODE REVIEW

File: components/Layout.tsx [hunk 1]
```diff
@@ -15,16 +17,18 @@ const WalletProvider = dynamic(
...
-      <WalletProvider>
-        <Logo />
-        <section id="dashboard">
-          <Header />
-          <Navbar />
-          <main> {children}</main>
-        </section>
-        <Footer />
-      </WalletProvider>
+      <QueryClientProvider client={queryClient}>
```
Comment (tompere) [inline components/Layout.tsx:22]: move along with all other contexts (top-level app)

File: components/List.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+} from '@mui/material';
+
+export const List = (props: { arr: any[] }) => {
+  return (
+    <div style={{ margin: '0.5rem' }}>
+      <TableContainer component={Paper}>
+        <Table sx={{ minWidth: 650 }} aria-label="simple table">
+          <TableHead>
+            <TableRow>
+              {Object.keys(Object.assign({}, ...props.arr)).map((val, id) => (
```
Comment (tompere) [inline components/List.tsx:19]:
```
Object.keys(props.arr)
```

File: package.json [hunk 3]
```diff
@@ -12,7 +12,9 @@
     "@emotion/react": "^11.10.5",
     "@emotion/styled": "^11.10.5",
     "@mui/material": "^5.11.0",
+    "@pankod/refine-antd": "^4.7.0",
```
Comment (tompere) [inline package.json]: removed `@pankod/refine-antd`

File: components/ka-ching/cash-register/CreateCashBox.tsx [hunk 4]
```diff
@@ -1,88 +1,277 @@
...
+        const transactionList = new Transaction({
+          blockhash: blockhash,
+          lastValidBlockHeight,
+          feePayer: cashier.publicKey!,
         });
-        const cashBoxTransaction = await cashBoxTx(cashRegId);
 
-        setCashBoxPubKey(cashBoxTransaction.instructions[0].keys[3].pubkey);
+        const arr: cashBoxModel[] = [];
+        let cashRegistrId: string = '';
```
Comment (tompere) [inline components/ka-ching/cash-register/CreateCashBox.tsx]: rewrite entire loop:
```ts
// consider moving to top-level module/file
const performTxns = async (cashboxModel) => { /* do your thing */ } 
const trasferTxns = await Promise.all(cashboxModels.map((cashboxModel) => performTxns(cashboxModel)));
await cashier.sendTransaction(transactionList, connection);
setAllTransfers(trasferTxns);
```

File: components/List.tsx (hunk 2 above)
Comment (BarLisML) [reply components/List.tsx:19]: If I use this "Object.keys(Object.assign({}, ...props.arr)).map((val, id)", then the names of the columns appear in the headers
But if I use this "Object.keys(props.arr).map((val, id)" then I get numbers in the names of the columns

File: components/Layout.tsx (hunk 1 above)
Comment (BarLisML) [reply components/Layout.tsx:22]: The main page that appears after the root top level page is the layout page
The providers are there for clean code in the root top level page

File: package.json (hunk 3 above)
Comment (BarLisML) [reply package.json]: removed

File: components/ka-ching/cash-register/CreateCashBox.tsx (hunk 4 above)
Comment (BarLisML) [reply components/ka-ching/cash-register/CreateCashBox.tsx]: done

**Review by tompere — APPROVED** [review]

**Pull request merged**